layer/layer_data.h

```diff
@@ -42,6 +42,87 @@ std::string CsvCat(Args&&... args) {
   return absl::StrJoin(std::forward_as_tuple(std::forward<Args>(args)...), ",");
 }
 
+// Base class for Vulkan dispatchable handle wrappers. Exposes the underlying
+// key to derived classes and implements basic operations like key comparisons
+// and hash.
+template <typename ConcreteKeyT>
+class DispatchableKeyBase {
+ public:
+  bool operator==(const ConcreteKeyT& other) const {
+    return key_ == other.key_;
+  }
+  bool operator!=(const ConcreteKeyT& other) const {
+    return key_ != other.key_;
```

Comment:
```suggestion
    return !operator==(other);
```

Reply:
Comparisons are not customizable by derived classes, so I don't think that this brings much values. I find the current version more readable, so I'll keep it.